skbio/util/_testing.py

```diff
@@ -8,6 +8,53 @@
 
 import os
 import inspect
+from nose import core
+from nose.tools import nottest
+from future.utils import PY3
+
+
+@nottest
+class TestRunner(object):
+    """Simple wrapper class around nosetests functionality.
+
+    Parameters
+    ----------
+    filename : __file__ attribute passed in from the caller. This tells the
+        tester where to start looking for tests.
+
+    Attributes
+    ----------
+    argv
+    test_dir
+
+    Notes
+    -----
+    The primary purpose of this class is to create an interface which users
+    of scikit-bio can use to run all of the built in tests. Normally this
+    would be done by invoking nosetests directly from the command line, but
+    scikit-bio needs several additional options which make the command long
+    and ugly. This class invokes nose with the required options.
+
+    """
+    def __init__(self, filename):
+        self._filename = filename
+        self._test_dir = os.path.dirname(filename)
+
+    def test(self):
+        """Performs the actual running of the tests.
+        """
+        # NOTE: it doesn't seem to matter what the first element of the argv
+        # list is, there just needs to be something there.
+        argv = [self._filename, '-I DO_NOT_IGNORE_ANYTHING']
+        if not self.is_py3():
```

Comment:
We could. I wanted to keep it abstracted since there were a couple different methods of determining the Python version put forward. I don't have strong feelings either way though.